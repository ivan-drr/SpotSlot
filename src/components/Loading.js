import React, { Component } from 'react';
import '../styles/Loading.css';

class Loading extends Component {

  constructor(props) {
    super(props);
    this.typeWriter = this.typeWriter.bind(this);
  }

  typeWriter(str, speed, id, iterator, intervals) {
    if (document.getElementById(id) !== null) {
      const element = document.getElementById(id);

      setTimeout(() => {
        if (iterator < str.length) {
          element.innerHTML += str.charAt(iterator);
          iterator++;

          setTimeout(this.typeWriter(str, speed, id, iterator), speed);

          return true;
        } else return false;
      }, 1000/speed*2);

    }
  }

  render() {
    let iterator = 0;

    if (this.props._isFetch === 'error') {
      setTimeout(() => {
        this.typeWriter(' âœ˜ Error connecting to API', 70, 'errorText', iterator);
      }, 0);
      return (
        <div id="loading" className="text-danger fixed-top">
          <span id="errorText" className="ml-3"></span>
        </div>
      );
    } else if (!this.props._isFetch) {
      setTimeout(() => {
        this.typeWriter(' Loading...', 70, 'loadingText', iterator);
      }, 0);
      return(
        <div id="loading" className="text-primary fixed-top">
          <div className="spinner-border text-primary" role="status"></div>
          <span id="loadingText" className="ml-3"></span>
        </div>
      );
    } else return null;
  }

}

export default Loading;
