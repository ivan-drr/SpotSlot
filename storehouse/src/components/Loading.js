import React, { Component } from 'react';
import '../styles/Loading.css';

class Loading extends Component {

  constructor(props) {
    super(props);
    this.typeWriter = this.typeWriter.bind(this);
  }

  typeWriter(str, speed, id, iterator) {
    if (document.getElementById(id) !== null) {
      var element = document.getElementById(id);

      setTimeout(() => {
        if (iterator < str.length) {
          element.innerHTML += str.charAt(iterator);
          iterator++;
          setTimeout(this.typeWriter(str, speed, id, iterator), speed);
        }
      }, 1000/speed*2);
    }
  }

  render() {
    var iterator = 0;

    if (this.props._isFetch === 'error') {
      document.getElementById('loadingText').innerHTML = '';
      this.typeWriter(' âœ˜ Error connecting to API', 70, 'loadingText', iterator);
      return (
        <div id="loading" className="text-danger fixed-top">
          <span id="loadingText" className="ml-3"></span>
        </div>
      );

    } else if (!this.props._isFetch) {
      this.typeWriter(' Loading...', 70, 'loadingText', iterator);
      return(
        <div id="loading" className="text-primary fixed-top">

          <div className="spinner-border text-primary" role="status"></div>
          <span id="loadingText" className="ml-3"></span>
        </div>
      );
    } else return null;
  }

}

export default Loading;
