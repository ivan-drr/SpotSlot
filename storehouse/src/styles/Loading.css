#loading {
  margin-top: 8%;
}
