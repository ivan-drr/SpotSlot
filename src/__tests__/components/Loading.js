import React from 'react';
import ReactDOM from 'react-dom'
import Enzyme, { shallow, render } from 'enzyme';
import Loading from '../../components/Loading';

beforeAll(() => {
  jest.disableAutomock();
});

beforeEach(() => {
  const element = document.getElementById('loadingText');
  if (element !== null) element.parentNode.removeChild(element);
});

describe('Unit Tests', () => {
  describe('Render', function () {

    it('should render correctly', () => {
      const loading = render(<Loading/>);
      expect(loading).toMatchSnapshot();
    });

    it('should render the error message', function () {
      const TEXT = ' ✘ Error connecting to API';
      const SPEED = 70;
      const ID = 'loadingText';
      //document.body.innerHTML = '<span id="'+ID+'"></span>';

      const loading = shallow(<Loading _isFetch={'error'}/>);

      setTimeout(() => {
        expect(document.getElementById(ID).textContent).toEqual(' ✘ Error connecting to API');
      }, SPEED * TEXT.length);
    });

    it('should return null on render call without DOM target defined', function () {
      const LoadingInstance = ReactDOM.render(<Loading _isFetch={true}/>, document.createElement('div'));
      expect(LoadingInstance.render()).toBe(null);
    });

    const LoadingInstance = ReactDOM.render(<Loading />, document.createElement('div'));

    describe('typeWriter', function () {
      it('should write on the DOM element', function () {
        var iterator = 0;
        const TEXT = 'test';
        const SPEED = 70;
        const ID = 'loadingText';

        document.body.innerHTML = '<span id="'+ID+'"></span>';

        setTimeout(() => {
          expect(document.getElementById(ID)).toEqual('test');
        }, SPEED * TEXT.length);

      });

      it('should warn that the DOM element was not found', function () {
        var iterator = 0;
        expect(LoadingInstance.typeWriter('test', 70, 'not-found', iterator)).toEqual('element_not_found');
      });

      it('should call again it self as many times as length of the given string', function () {
        var iterator = 0;
        const TEXT = 'test';
        const SPEED = 70;
        const ID = 'loadingText';

        document.body.innerHTML = '<span id="'+ID+'"></span>';

        LoadingInstance.typeWriter(TEXT, SPEED, ID, iterator);

        setTimeout(() => {
          expect(LoadingInstance.typeWriter).toHaveBeenCalledTimes(TEXT.length);
        }, SPEED * TEXT.length);
      });

    });
  });

});
