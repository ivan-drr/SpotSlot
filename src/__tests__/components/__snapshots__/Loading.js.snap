// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Tests Render should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "text-primary fixed-top",
      "id": "loading",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "spinner-border text-primary",
          "role": "status",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ml-3",
            "id": "loadingText",
          },
          "children": Array [],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "id": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ml-3",
          "id": "loadingText",
        },
        "children": Array [],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "spinner-border text-primary",
            "role": "status",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "role": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "role": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "id": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "id": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
